module/zfs/dmu.c

```diff
@@ -1132,24 +1131,8 @@ dmu_read_uio_dnode(dnode_t *dn, uio_t *uio, uint64_t size)
 		bufoff = uio->uio_loffset - db->db_offset;
 		tocpy = MIN(db->db_size - bufoff, size);
 
-		if (xuio) {
-			dmu_buf_impl_t *dbi = (dmu_buf_impl_t *)db;
-			arc_buf_t *dbuf_abuf = dbi->db_buf;
-			arc_buf_t *abuf = dbuf_loan_arcbuf(dbi);
-			err = dmu_xuio_add(xuio, abuf, bufoff, tocpy);
-			if (!err) {
-				uio->uio_resid -= tocpy;
-				uio->uio_loffset += tocpy;
-			}
-
-			if (abuf == dbuf_abuf)
-				XUIOSTAT_BUMP(xuiostat_rbuf_nocopy);
-			else
-				XUIOSTAT_BUMP(xuiostat_rbuf_copied);
-		} else {
-			err = uiomove((char *)db->db_data + bufoff, tocpy,
-			    UIO_READ, uio);
-		}
+		err = uiomove((char *)db->db_data + bufoff, tocpy,
+		    UIO_READ, uio);
```

Comment:
This code was intentionally left in place even though it's dead.  The rest of the xuio code was wrapped in a `#ifdef HAVE_UIO_ZEROCOPY` define which is never defined.  The idea at the time was to leave this code largely intact until it could be updated for Linux and renabled.  This block of code should get the same treatment.  You can also wrap the following helper functions a few lines above in `#ifdef HAVE_UIO_ZEROCOPY`

dmu_xuio_init 
dmu_xuio_fini 
dmu_xuio_add 
dmu_xuio_cnt 
dmu_xuio_arcbuf 
dmu_xuio_clear